core/src/main/scala/com/banno/kafka/producer/ProducerImpl.scala

```diff
@@ -89,7 +90,7 @@ case class ProducerImpl[F[_], K, V](p: Producer[K, V])(implicit
         record,
         { (rm, e) => callback(Option(e).toLeft(rm).toTry) },
       )
-    ).map(jf => F.delay(jf.cancel(true)).void)
+    ).map(jf => F.interruptible(jf.cancel(true)).void)
```

Comment:
Sorry, I should've mentioned: we've already gone through all of the send-related methods, and those delays can remain.

Reply:
got it! ty.